models/models.js

```diff
@@ -86,12 +86,53 @@ exports.fetchComments = (articleId) => {
       }
    return connection.query(`SELECT * FROM articles WHERE article_id= $1`,[articleId])
      .then(({rowCount}) =>{
-      console.log(rowCount)
       if (rowCount === 0){
         return Promise.reject({status: 404, msg: 'The article is not found'});
       }
       return rows;
      })
     });
-
   };
+
+  //=======================================================
+
+exports.addComment = (username, body, articleId) => {
+  //console.log(username, body, articleId)
+  if (isNaN(+articleId)) {
+    return Promise.reject({ status: 400, msg: "The article_id has to be a number" });
+  }
+  if (typeof username !== "string" || typeof body !== "string") {
+    return Promise.reject({ status: 400, msg: "Bad Request",
+    });
+  }
+
+   if (body === undefined) {
+    return Promise.reject({ status: 400, msg: "Bad Request" });
+   }
+   if (username === undefined) {
```

Comment:
You can make the code DRY by combining this if statement with line 109 and 104, because their Promise.rejects are all the same anyway! :)

Reply:
Hi Karla
One of the mentor has approved it

Reply:
and I have already  merged it

Reply:
I will make the code DRY by combining this if statement with line 109 and 104, because their Promise.rejects are all the same anyway! :)